### PR TITLE
In check

### PR DESCRIPTION
**Pull request by KazimieraLuce**
I had to change a few things within the bishop and rook models as well as the piece model to get the tests to pass. If everyone could please look and let me know if I messed your code up too badly, then we can figure out how to fix it together. Thanks.

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -4,40 +4,68 @@ class Game < ApplicationRecord
...
+     Bishop.create(game_id: id, x_space: 5, y_space: 7, color: "white")
+
+     Queen.create(game_id: id, x_space: 3, y_space: 7, color: "white")
+     King.create(game_id: id, x_space: 4, y_space: 7, color: "white")
+   end
 
-   Knight.create(game_id: id, x_space: 1, y_space: 7, color: "white")
-   Knight.create(game_id: id, x_space: 6, y_space: 7, color: "white")
+  def check?(color)
+    black_king = game.pieces.find_by(color: "black", type: "King")
```
Comment (ZoeBKramer) [inline app/models/game.rb:44]: make sure you add the game id here too! Want to make sure it's finding only the black kings in this game, not all of the black kings in every game ever ;)

**Review by ZoeBKramer — CHANGES_REQUESTED** [review]

File: app/models/game.rb (hunk 1 above)
Comment (ZoeBKramer) [reply app/models/game.rb:44]: totally didn't see the game at the beginning, that should work. Sorry!

File: app/models/game.rb [hunk 2]
```diff
@@ -4,40 +4,68 @@ class Game < ApplicationRecord
...
+    white_king = game.pieces.find_by(color: "white", type: "King")
+    
+    black_piece = game.pieces.find_by(color: "black", type: "Piece")
```
Comment (ZoeBKramer) [inline app/models/game.rb]: I see that you're trying to fetch all the pieces, however Piece is not a type. Only King, Queen, Rook, Bishop, etc. You should replace `find_by` with `where` (it returns an array of pieces where as `find_by` only returns a single object.) When you put `where`, you are able to remove the type and just have the color passed in.

**Review by ZoeBKramer — CHANGES_REQUESTED** [review]

File: app/models/game.rb (hunk 2 above)
Comment (KazimieraLuce) [reply app/models/game.rb]: Thank you so much. I knew I should have looked at it again this morning before submitting. 🤣 Will fix and submit tomorrow.

**Review by BrandonJohnson65 — APPROVED** [review]

**Review by BrandonJohnson65 — APPROVED** [review]

**Comment by KazimieraLuce** [comment]
Hi Zoe. When you have a moment, would you review the new `check?` method and let me know what you think? thanks.

**Review by ZoeBKramer — APPROVED** [review]

**Pull request merged**